File: evolab-frontend/EvoLabAsAService/src/types/job.ts
import { EvolutionStatus } from './evolution';

export interface Job {
    id: number;
    projectId: number;
    status: EvolutionStatus;
    containerId: string | null;
    startedAt: string | null; // ISO 8601 Instant
    finishedAt: string | null; // ISO 8601 Instant
    bestSolution: string | null;
    executionLogs: string | null;
    createdAt: string; // ISO 8601 Instant
}

File: evolab-backend/domain/src/main/kotlin/com/example/evolab/domain/job/Job.kt
package com.example.evolab.domain.job

import com.example.evolab.domain.evolution.EvolutionStatus
import java.time.Instant

data class Job(
    val id: Int,
    val projectId: Int,
    val status: EvolutionStatus,
    val containerId: String?,
    val startedAt: Instant?,
    val finishedAt: Instant?,
    val bestSolution: String?,
    val executionLogs: String?,
    val createdAt: Instant,
    val failureReason: String? = null,
)

